ibm/service/secretsmanager/resource_ibm_sm_public_certificate.go

```diff
@@ -887,3 +930,222 @@ func ConcreteListToStringSlice(elements []interface{}, commonName string) ([]str
+
+func createOrUpdateAkamaiChallengeRecordSet(config edgegrid.Config, zone string, txtRecordName string, ttl int, rdata []string, method string) diag.Diagnostics {
+	type TXTRecordSet struct {
+		Name  string   `json:"name"`
+		Type  string   `json:"type"`
+		TTL   int      `json:"ttl"`
+		Rdata []string `json:"rdata"`
+	}
+
+	recordSetBody := TXTRecordSet{
+		Name:  txtRecordName,
+		Type:  "TXT",
+		TTL:   ttl,
+		Rdata: rdata,
+	}
+
+	jsonBody, err := json.Marshal(recordSetBody)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error setting body for akamai request: %s", err))
+	}
+	req, err := client.NewRequest(config, method, fmt.Sprintf("/config-dns/v2/zones/%s/names/%s/types/TXT", zone, txtRecordName), bytes.NewReader(jsonBody))
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating akamai request: %s", err))
+	}
+	res, err := client.Do(config, req)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error in akamai request: %s", err))
+	}
+	if res.StatusCode != 201 && res.StatusCode != 200 {
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			fmt.Printf("Error reading response: %s\n", err.Error())
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("error from akamai in '%s' request: %s", method, string(body)))
+	}
+	return nil
+}
+
+func findAllTxtRecordValuesForDomain(domainItem string, txtRecordName string, secret *secretsmanagerv2.PublicCertificate, successfullySetChallengeDomains map[string]string) ([]string, diag.Diagnostics) {
+	challenges := secret.IssuanceInfo.Challenges
+	var txtRecordValues []string
+
+	for _, challengesItem := range challenges {
+		if *challengesItem.TxtRecordName == txtRecordName {
+			if _, exists := successfullySetChallengeDomains[*challengesItem.TxtRecordValue]; !exists {
+				txtRecordValues = append(txtRecordValues, *challengesItem.TxtRecordValue)
+				successfullySetChallengeDomains[*challengesItem.TxtRecordValue] = *challengesItem.Domain
+			} else { // if the txtRecordValue exists --> this txtRecordName (domain) has already been checked --> no need to continue
+				return txtRecordValues, nil
+			}
+		}
+	}
+
+	if len(txtRecordValues) > 0 {
+		return txtRecordValues, nil
+	}
+	return nil, diag.FromErr(fmt.Errorf("failed to find a challenge for the domain: %s", domainItem))
+}
+
+func findTxtRecordValuesDifferences(akamaiValues, challengesValues []string) []string {
+	var differences []string
+
+	if akamaiValues != nil {
+		set := make(map[string]bool)
+		for _, value := range akamaiValues {
+			set[value] = true
+		}
+
+		for _, challengeValue := range challengesValues {
+			stringChallengeValue := "\"" + challengeValue + "\""
+			if !set[stringChallengeValue] {
+				differences = append(differences, challengeValue)
+			}
+		}
+	}
+
+	return differences
+}
+
+func getZone(currentZone string, originalDomain string, config edgegrid.Config) (string, diag.Diagnostics) {
+	req, err := client.NewRequest(config, "GET", fmt.Sprintf("/config-dns/v2/zones/%s", currentZone), nil)
```

Comment:
As general practise the code which handles all REST will be on SDK side . Do we have any SDK which handles this?

Reply:
@hkantare No. They don't have a GO sdk we can use.